File: 프로그래머스/2/42842. 카펫/카펫.py
def solution(brown, yellow):
    answer = []
    total = brown + yellow
    for i in range(1, total):
        if total%i == 0:
            j = total//i
            num = (i+j)*2-4
            if num == brown:
                answer = [j,i]
                break
            
    return answer

File: 프로그래머스/2/42842. 카펫/카펫.js
function solution(brown, yellow) {
    var answer = [];
    let total = brown + yellow;

    for (let i = 1; i <= total; i++) {
        if (total % i === 0) {
            let j = total / i;
            let num = (i + j) * 2 - 4;
            if (num === brown) {
                answer = [j, i];
                break;
            }
        }
    }

    return answer;
}